src/Symfony/Component/Validator/DependencyInjection/AddAutoMappingConfigurationPass.php

```diff
@@ -27,12 +27,12 @@ class AddAutoMappingConfigurationPass implements CompilerPassInterface
      */
     public function process(ContainerBuilder $container)
     {
-        if (!$container->hasParameter('validator.auto_mapping') || !$container->hasDefinition('validator.builder')) {
+        $config = $container->hasParameter('validator.auto_mapping') ? $container->getParameter('validator.auto_mapping') : null;
+
+        if (!$config || !$container->hasDefinition('validator.builder')) {
```

Comment:
Should still remove the parameter from container IMHO:

```suggestion
        if (!$config || !$container->hasDefinition('validator.builder')) {
            $container->getParameterBag()->remove('validator.auto_mapping');
            
```